android/src/main/java/com/iterable/reactnative/Serialization.java

```diff
@@ -31,4 +50,154 @@ static IterableInAppDeleteActionType getIterableDeleteActionTypeFromInteger(@Nul
             return IterableInAppDeleteActionType.values()[actionType];
         }
     }
-}
+
+    static JSONArray getInAppMessages() {
+        List<IterableInAppMessage> inappMessages = IterableApi.getInstance().getInAppManager().getMessages();
+
+        JSONArray inappMessagesJson = new JSONArray();
+
+        for (IterableInAppMessage message : inappMessages) {
+            JSONObject messageJson = new JSONObject();
+            try {
+                messageJson.putOpt("messageId", message.getMessageId());
+                //TODO: Incorporate public getter method for campaignId in next Android SDK release
+                //messageJson.putOpt("campaignId", message.getcampignId());
+                messageJson.putOpt("trigger", RNIterableInternal.getTriggerType(message));
+                messageJson.putOpt("createdAt", (int) message.getCreatedAt().getTime() / 1000);
```

Comment:
Getting a wrong aura around this... 🙇‍♂️

Reply:
I'd wrap this stuff into a method since you're doing the same a few lines later too.

Reply:
Also type casting to (int) is changing the epoch date value. Keeping it `long` and assigning to JSON.